include/irr/asset/filters/CSwizzleAndConvertImageFilter.h

```diff
@@ -28,13 +27,17 @@ namespace impl
 {
 
 
-template<typename Swizzle>
+template<typename Swizzle, typename Dither>
 class CSwizzleAndConvertImageFilterBase : public CMatchedSizeInOutImageFilterCommon
 {
 	public:
 
 		class CState : public CMatchedSizeInOutImageFilterCommon::state_type, public Swizzle
 		{
+			public:
+				Dither dither;
+				using DitherState = typename Dither::state_type;
+				DitherState* ditherState = nullptr;					//! To use dithering in the filter - allocate, create and make the dither state pointer valid
```

Comment:
there should be some sort of conditional (`enable_if`) declaration of these member variables

Basically if I'd set `Dither=void` it means no dither decided at compile time.

Reply:
see how there's a separate state member for `PolymorphicSwizzle` and there isn't when the swizzle is not polymorphic

Reply:
there is a IdentityDither class now according to DC conversation and it works fine (at compile time)